core/src/main/scala/io/github/gaelrenoux/tranzactio/DatabaseServiceBase.scala

```diff
@@ -15,43 +15,33 @@ abstract class DatabaseServiceBase[Connection: Tag](connectionSource: Connection
   def connectionFromJdbc(connection: => JdbcConnection)(implicit trace: Trace): ZIO[Any, Nothing, Connection]
 
   override def transaction[R, E, A](zio: => ZIO[Connection with R, E, A], commitOnFailure: => Boolean = false)
-    (implicit errorStrategies: ErrorStrategiesRef, trace: Trace): ZIO[R, Either[DbException, E], A] =
-    ZIO.environmentWithZIO[R] { r =>
```

Comment:
This `R` can be provided with `.provideSomeEnvironment[R](_ ++ ZEnvironment(connection))`